test/unittests/commands/xmlupload/test_check_consistency_with_ontology.py

```diff
@@ -0,0 +1,314 @@
+                    <resptr-prop name=":hasResource2">
+                        <resptr permissions="prop-default">resB</resptr>
+                    </resptr-prop>
+                    <resptr-prop name=":hasResource3">
+                        <resptr permissions="prop-default">resA</resptr>
+                    </resptr-prop>
+                </resource>
+        </knora>"""
+    )
+    expected_classes = [["resA", ":TestThing1"], ["resB", ":TestThing2"], ["resC", ":TestThing2"]]
+    expected_properties = [["resA", ":hasResource1"], ["resC", ":hasResource2"], ["resC", ":hasResource3"]]
+    res_classes, res_properties = _get_all_classes_and_properties(test_ele)
+    assert unordered(res_classes) == expected_classes
+    assert unordered(res_properties) == expected_properties
+
+
+def test_get_all_class_types_and_ids() -> None:
+    test_ele = etree.fromstring(
+        """<knora shortcode="4123" default-ontology="test">
+                <resource label="resA" restype=":TestThing1" id="resA" permissions="res-default">
+                </resource>
+                <resource label="resB" restype=":TestThing2" id="resB" permissions="res-default">
+                </resource>
+                <resource label="resC" restype=":TestThing2" id="resC" permissions="res-default">
+                </resource>
+        </knora>"""
+    )
+    expected = [["resA", ":TestThing1"], ["resB", ":TestThing2"], ["resC", ":TestThing2"]]
+    assert unordered(_get_all_class_types_and_ids(test_ele)) == expected
+
+
+def test_get_all_property_names_and_resource_ids() -> None:
+    test_ele = etree.fromstring(
+        """<resource label="resC" restype=":TestThing2" id="resC" permissions="res-default">
+                <resptr-prop name=":hasResource2">
+                    <resptr permissions="prop-default">resB</resptr>
+                </resptr-prop>
+                <resptr-prop name=":hasResource3">
+                    <resptr permissions="prop-default">resA</resptr>
+                </resptr-prop>
+            </resource>"""
+    )
+    expected = [["resC", ":hasResource2"], ["resC", ":hasResource3"]]
+    assert unordered(_get_all_property_names_and_resource_ids(test_ele)) == expected
+
+
+def test_find_problems_in_classes_and_properties() -> None:
+    onto_check_info = OntoCheckInformation(
+        default_ontology_prefix="test",
+        save_location=Path(""),
+        onto_lookup={
+            "test": Ontology(classes=["classA", "classB"], properties=["propA", "propB"]),
+            "knora-api": Ontology(classes=["knoraClassA"], properties=["knoraPropA"]),
+        },
+    )
+    classes = [["idA", "knoraClassA"]]
+    properties = [["idA", "knora-api:knoraPropA"]]
+    assert not _find_problems_in_classes_and_properties(
+        classes, properties, onto_check_info
+    )  # type: ignore[func-returns-value]
+
+
+class TestDiagnoseClass:
+    @staticmethod
```

Comment:
this is purely cosmetic - but I think to my eye it would look less cluttered to add `self` to each of those test functions (and not use it), than to have the `@staticmethod` decorator everywhere